### PR TITLE
LPS-186330 Listing Organizations as a System Object

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue:
https://issues.liferay.com/browse/LPS-186330
https://issues.liferay.com/browse/LPS-186839
https://issues.liferay.com/browse/LPS-186840

### CODE REVIEW

**Review by p-albuquerque — APPROVED** [review]

**Comment by mateussgn** [comment]
Hey @MarinhoFeliphe, @p-albuquerque, QA, and CI reviews were already made and approved.
Could you double-check the code review?

File: modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+	@Override
+	public List<ObjectField> getObjectFields() {
+		return Arrays.asList(
+			createObjectField(
+				"Text", "comments", "String", "comments", "comments", false,
+				true),
+			createObjectField(
+				"Text", "name", "String", "name", "name", true, true));
+	}
```
Comment (MarinhoFeliphe) [inline modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java:127]: @mateussgn, 

could you please replace `"Text"` to `ObjectFieldConstants.BUSINESS_TYPE_TEXT` and `"String"` to `ObjectFieldConstants.DB_TYPE_STRING`?

it's just to [avoid magic strings](https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings) it's a good practice.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+		return builder.checkPermissions(
+			false
+		).preferredLocale(
+			user.getLocale()
+		).user(
+			user
+		).build();
+	}
+
```
[the remark this answers is not in the document]
Comment (mateussgn) [reply modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java:196]: Sure!
Thanks for the tip!
I'll check if there is any other method missing implementation.

File: modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java (hunk 1 above)
Comment (mateussgn) [reply modules/apps/organizations/organizations-service/src/main/java/com/liferay/organizations/internal/object/system/OrganizationSystemObjectDefinitionManager.java:127]: Sure!
Thanks for the tip!

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
@MarinhoFeliphe , could you review it again?

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/2265

**Pull request closed without merging**